polaris-service/src/test/java/org/apache/polaris/service/PolarisApplicationIntegrationTest.java

```diff
@@ -341,7 +341,7 @@ public void testIcebergListNamespacesNestedNotFound() throws IOException {
                   sessionCatalog.listNamespaces(
                       sessionContext, Namespace.of("top_level", "whoops")))
           .isInstanceOf(NoSuchNamespaceException.class)
-          .hasMessage("Namespace does not exist: top_level.whoops");
+          .hasMessage("Namespace does not exist: top_level%1Fwhoops");
```

Comment:
Checking this.

Reply:
found this change : https://github.com/apache/iceberg/commit/5fc1413a5efc4419ccc081f3031325f107ccddab looks like we changes the seperator, 
pending to make it configurable https://github.com/apache/iceberg/pull/10877

Reply:
There's no more intent to make the separator configurable. At least, that's the latest I know. There was a big discussion around this topic in August this year.
IMHO, making the separator configurable makes things overly complicated.

Reply:
Is this the response in the error message now? I think the dot in the error is much more user friendly

Reply:
This is interesting, as Iceberg still uses dot to join different level of namespaces, https://github.com/apache/iceberg/blob/09634857e4a1333f5dc742d1dca3921e9a9f62dd/api/src/main/java/org/apache/iceberg/catalog/Namespace.java#L97-L97. 
In this case, `Namespace.of("top_level", "whoops")` should be referred to `top_level.whoops`, unless the rest util somehow squash two levels together with the "%1F".
@singhpk234 , could you take a look?

Reply:
I believe the server is missing a call to `RESTUtil.decodeNamespace(..)`, hence you're getting the encoded namespace here

Reply:
changing https://github.com/apache/polaris/blob/ce3634c80f5b97daa8410be49c1fd4d2ff363e74/polaris-service/src/main/java/org/apache/polaris/service/catalog/IcebergCatalogAdapter.java#L183 to `RESTUtil.decodeNamespace(namespace)` should fix this

Reply:
didn;t work that great https://github.com/apache/polaris/pull/442/commits/7bfa42d1b3ea8e0d06e536a9369ad8d736e7739a

Tracking ticket : 
https://github.com/apache/iceberg/issues/11539

Reply:
Hi @singhpk234 , What's the error in https://github.com/apache/polaris/commit/7bfa42d1b3ea8e0d06e536a9369ad8d736e7739a? Couldn't get it once the new pipeline started.

Reply:
oops looks like it was a style miss : https://github.com/apache/polaris/actions/runs/11859822454/job/33055322598
let me rerun style and test locally first

Reply:
It seems the revert PR (https://github.com/apache/iceberg/pull/11574) is targeting Iceberg 1.7.1. Should we wait for the 1.7.1 release to avoid unnecessary back-and-forth?

Reply:
Let me see, when the pr get merged and we have nightly artifact published, I can try reverting my change to correct the test and run our polaris suite with that ? wdyt

Reply:
can confirm its passing now, I have pointed to 1.8.0 artifact from nigthly